### ISSUE TITLE
[Data Hookup] - Show local application errors on Application Load 

### ISSUE THREAD

**Issue by rosibaj** [icgc-argo/dac-ui#163]
## Goal
A user can save "bad data" as part of form entry.  While the data is saved, the BE keeps the state as incorrect and records the errors that are in that section. 

When an application is loaded, the local validations need to be triggered after load to show the errors in a section to users. 

## Example Data
- https://dac.dev.argo.cancercollaboratory.org/applications/DACO-42
- You can see in the representative body sectoin, i have an `INCOMPLETE` state and `ERROR` in meta
```
        "representative": {
            "address": {
                "building": "",
                "cityAndProvince": "",
                "country": "",
                "postalCode": "",
                "streetAddress": ""
            },
            "addressSameAsApplicant": true,
            "info": {
                "firstName": "Sarumon",
                "googleEmail": "sarumon@example.com",
                "institutionEmail": "@example.com",     <-------------⚠️⚠️⚠️
                "displayName": "Sarumon the White",
                "institutionWebsite": "https://www.facit.on.ca",
                "lastName": "the White",
                "middleName": "",
                "positionTitle": "Warlock",
                "primaryAffiliation": "",
                "suffix": "string",
                "title": "M"
            },
            "meta": {
                "errorsList": [    <-------------⚠️⚠️⚠️ 
                    {
                        "field": "institutionEmail",
                        "message": "Value is not a valid email"
                    },
                    {
                        "field": "primaryAffiliation",
                        "message": "field primaryAffiliation is required"
                    },
                    {
                        "field": "primaryAffililation",
                        "message": "Primary Affiliation must be the same as the Applicant"
                    }
                ],
                "status": "INCOMPLETE"
            }
        },
```

## Expected Outcomes
- when an application ls loaded, all fields should display the contents that are recorded in the BE
- For each field that is listed in the `meta.errorsList` trigger the local validation so that the error shows on that field
- In this example above, this is how the display should look: 
![image](https://user-images.githubusercontent.com/15824252/123656991-64ef4700-d7fe-11eb-93e5-3fe01bca1e32.png)

Note: this does not include the extended error UI for modals (#142) or Primary Affiliation (#54)

**Comment by caravinci** [comment]
As part of this ticket, I'll also ensure the word counter tracks the number upon loading data from BE.
![image.png](https://images.zenhubusercontent.com/5defce3f99bcf0b1b8a98d24/4a911a3a-111e-4e46-b891-d66866f3bad1)

**Comment by caravinci** [comment]
implementation completed as per spec. Some fields are not being tagged as invalid in the API payload (e.g. the terms checkbox).
For some few fields, there may be some fine tuning needed (tests pending) as I had to compensate for the field names in the `errorsList` not matching the convention given by the data schema.

That said, most fields' validation works as intended, and this should be a good MVP level of error handling.

I've also attached the PR for the textarea counter issue mentioned in my previous comment, and I'll make sure the improvement will be added as soon as the new package version has been published.
